portfolio/src/main/java/com/google/sps/servlets/FactsServlet.java

```diff
@@ -0,0 +1,48 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.sps.data.FactsData;
+
+/** Handles requests sent to the /facts URL. Try running a server and navigating to /hello! */
+@WebServlet("/facts")
+public class FactsServlet extends HttpServlet {
+
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
I'm not sure which line of code in this method potentially throws the IOException, but it's a good habit to double check that the IOException that your method could throw contains enough information for you to debug the issue when the exception does occur.

If the IOException is too generic, you can put the method that throws the exception in a try/catch block and generate your own exception to send for easier debugging. For example, if `response.getWriter()` could potentially throw an exception, we might update this code to be:

```
// Send the JSON as the response
response.setContentType("application/json;");
try {
    response.getWriter().println(json);
} catch (IOException e) {
    throw new IOException("IOException occurred while getting the /facts URL.", e);
}
```
This code passes along the IOException that would have been thrown by `response.getWriter()`, but it lets us add a little more information to it. No need to add this to this commit, but thought you might be interested in how to better debug this potential Exception in the future!

Reply:
I see. I have never done error handling (throw exception using try/catch) in the protfolio project yet, but this is really helpful when the project code gets large and hard to debug. Thanks Gina for pointing that out!